lib/std/io/early_eof_reader.zig

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2020 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+const std = @import("../std.zig");
+const io = std.io;
+const assert = std.debug.assert;
+const testing = std.testing;
+
+pub fn EarlyEOFReader(comptime ReaderType: type) type {
+    return struct {
+        child_stream: ReaderType,
```

Comment:
Nit, this is a reader and not a stream so `inner_reader` may be a better name.

Reply:
FWIW I copied the field name from `CountingWriter`.

Reply:
I didn't write/review that code :^), I'm [sticking](https://github.com/ziglang/zig/blob/2b7781d82ad8d2234b89257676670957e005f214/lib/std/compress/zlib.zig#L27) to this naming convention.
My understanding is that a `Stream` is the backing object out of which you pull a `Reader` or a `Writer`.